src/Listeners/Plugin.php

```diff
@@ -44,7 +56,212 @@ public function deactivated( $plugin, $network_wide ) {
 		$data = array(
 			'plugin'       => $plugin,
 			'network_wide' => $network_wide,
+			'plugins'      => $this->collect_plugin( $plugin, false ),
 		);
 		$this->push( 'plugin_deactivated', $data );
 	}
+
+	/**
+	 * Plugin deleted
+	 *
+	 * @param string  $plugin Name of the plugin
+	 * @param boolean $deleted Whether the plugin deletion was successful
+	 * @return void
+	 */
+	public function deleted( $plugin, $deleted ) {
+		// abort if not successfully deleted
+		if ( !$deleted ) {
+			return;
+		}
+		
+		$slug = self::glean_plugin_slugname( $plugin );
+		$data = array(
+			'plugin' => $slug,
+		);
+		$this->push( 'plugin_deleted', $data );
+	}
+
+	/**
+	 * Plugin install or update completed
+	 *
+	 * @param string  $upgrader_object Upgrader Object from upgrade hook
+	 * @param boolean $options Options from upgrade hook including type, action & plugins.
+	 * @return void
+	 */
+	public function updated( $upgrader_object, $options ) {
+		// bail if not plugin install or update
+		if ( 'plugin' !== $options['type'] ) {
+			return;
+		}
+		// set event type
+		if ( 'update' === $options['action'] ) {
+			$event_key = 'plugin_updated';
+
+			$plugins = [];
+			if ( isset( $options['plugins'] ) && is_array( $options['plugins'] ) ) {
+				foreach ( $options['plugins'] as $index => $pluginslug ) {
+					$plugin = $this->collect_plugin( 
+						$pluginslug, 
+						is_plugin_active( $pluginslug ),
+						false
+					);
+					array_push( $plugins, $plugin );
+				}
+			}
+			$data = array(
+				'plugins' => $plugins,
+			);
+		
+		} elseif ('install' === $options['action'] ) {
+			$event_key = 'plugin_installed';
+			// get all plugins - slug not returned for install actions
+			$plugins = $this->collect_plugins();
+			$data = array(
+				'plugins' => $plugins,
+			);
```

Comment:
Don't need to store it on `$plugins` here. Save the variable assignment.
```suggestion
			$data = array(
				'plugins' => $this->collect_plugins(),
			);
```

Reply:
Thanks, that's more clear too.